### ISSUE TITLE
"BaseClient" has no attribute "describe_volumes" (and others)

### ISSUE THREAD

**Issue by jlgoolsbee** [vemel/mypy_boto3_builder#101]
**Describe the bug**

When [creating low-level clients from resources](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/clients.html#creating-clients), it seems mypy is unaware of the available methods and attributes for that client object/class and also rejects attempts to override the client class when assigned to a variable. The example below uses the `describe_volumes` method for the EC2 client, but it seems applicable to at least any EC2 method (e.g., `describe_volumes_modifications`, `describe_instance_status`).

I tried to find an example / other specific way of doing this (creating a client from resource) that wouldn't trigger these errors (hence my attempt at overriding the class via a variable assignment); perhaps I'm missing something, but since creating clients in this way is a documented/established pattern in the boto3 docs, it seems like this should work without any hoops to jump through.

**To Reproduce**

1. Install 'boto3-stubs[ec2]'
2.  Run mypy on the following code sample

```python
#!/usr/bin/env python3

import boto3

ec2 = boto3.resource('ec2')

volumes = ec2.meta.client.describe_volumes(
    Filters=[{'Name': 'tag:Name', 'Values': ["somevalue"]}]
)
```

3. Actual mypy output

```
example.py:7: error: "BaseClient" has no attribute "describe_volumes"
Found 1 error in 1 file (checked 1 source file)
```

4. Attempt to override client class:

```python
#!/usr/bin/env python3

import boto3
from mypy_boto3_ec2.client import EC2Client

ec2 = boto3.resource('ec2')
ec2_client: EC2Client = ec2.meta.client

volumes = ec2_client.describe_volumes(
    Filters=[{'Name': 'tag:Name', 'Values': ["somevalue"]}]
)
```

5. Actual mypy output

```
example.py:7: error: Incompatible types in assignment (expression has type "BaseClient", variable has type "EC2Client")
Found 1 error in 1 file (checked 1 source file)
```

**Expected mypy-output**

None (I'd expect no errors for the first example; I'm not sure if the second example _should_ be possible).

**Additional context**

Mac OS 10.15.7, `boto3-stubs` installed via pip (1.18.2), `boto3` version 1.18.2

**Comment by vemel** [comment]
Hello! Thank you for the report.

Thanks for the report. Looks like I use `BaseClient` annotations for `EC2ServiceResource.meta.client` instead of `EC2Client`.
The solution would be to define `meta` separately for each submodule.

I am going to release a bugfix tomorrow. Meanwhile, you can initialize client directly `boto3.client('ec2')` or use explicit type annotations.

**Comment by vemel** [comment]
The latest `boto3-stubs` has proper typing for `resource.meta.client`.

Please update and let me know if it works for you as expected.

**Comment by jlgoolsbee** [comment]
Thanks for the quick response and fix! I can confirm this is resolved in boto3-stubs v1.18.4 🎉
